src/main/scala/Models.scala

```diff
@@ -1,21 +1,38 @@
+import spray.json._
+import DefaultJsonProtocol._
+import sangria.execution.deferred.HasId
+import sangria.schema._
+import sangria.macros.derive._
+
 case class Category(
   id: Int,
-  name: String
-) {
-  def styles(allStyles: List[Style]): List[Style] =
-    allStyles.filter(_.category_id == id)
+  name: String)
+
+object Category {
+  implicit val categoryHasId = HasId[Category, Int](_.id)
+  implicit val categoryFormat = jsonFormat2(Category.apply)
+
+  implicit val graphqlType: ObjectType[Repository, Category] = deriveObjectType(
+    ObjectTypeDescription("A category"),
+    AddFields(Field("styles", ListType(Style.graphqlType),
+      resolve = c => c.ctx.styleFetcher.deferRelSeq(c.ctx.styleByCategory, c.value.id))))
 }
```

Comment:
The domain models in this case still completely unaware of JSON serialization or GraphQL. I guess it's just very convenient way to pack all type-class instances for particular case class in its companion object. They are indeed co-located in a single file, but from application's perspective, they are sill loosely coupled.